### ISSUE TITLE
Change handling of potentially ambiguous method overloads

### ISSUE THREAD

**Issue by lukedegruchy** [cqframework/clinical_quality_language#1243]
See https://github.com/cqframework/clinical_quality_language/issues/1127

After this fix was merged, @brynrhodes suggested some fixes:

1. Leverage the OperatorMap to implement this check in `LibraryBuilder#resolveInvocation` instead of `validateAmbiguousOverloadedForwardDeclarationsSignatureNone()`, specifically, around the block that checks the SignatureLevel
2. Consider making this a warning and not an error
3. Consider checking this only if the function(s) in question are called, not if they merely exist in the CQL or in the library in question

**Comment by brynrhodes** [comment]
Note that per some internal discussion, the implementation here is just to change this logic:

https://github.com/cqframework/clinical_quality_language/blob/master/Src/java/cql-to-elm/src/main/java/org/cqframework/cql/cql2elm/LibraryBuilder.java#L1176

So that in the case that we have resolved an overload with a different invocation signature than the declaration signature and the signatureLevel is None, that we issue a warning, not an error.

The warning should say "Function invocation has a different signature than the declaration and the SignatureLevel is set to none. This may result in ambiguous function resolution at runtime, consider setting the SignatureLevel to higher than None."

Specifically, I'm trying to make sure the message doesn't say anything about forwards, because it's not about forwards, this happens regardless of whether the function is a forward declaration.

Also, the default options in the translator were changed to SignatureLevel.Overloads, and we should switch that back to None.
